.github/workflows/actions.yml

```diff
@@ -19,11 +15,26 @@ jobs:
         os: [ubuntu-latest, windows-latest, macos-latest]
     name: ${{ matrix.os }} - Java ${{ matrix.java }}
     steps:
-      - uses: actions/checkout@v2
+      - name: Check out repo
+        uses: actions/checkout@v2
+        # fetch unshallow to enable blame for Sonar
+      - name: Fetch unshallow
+        run: git fetch --prune --unshallow
       - name: Setup java
         uses: actions/setup-java@v1
         with:
           java-version: ${{ matrix.java }}
+      - name: Cache Gradle
+        uses: actions/cache@v1
+        with:
+          path: ~/.gradle/caches/
+          key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle*') }}
+          restore-keys: ${{ runner.os }}-gradle-
+      - name: Cache Sonar
```

Comment:
sonar only runs on ubuntu and only on one JDK - maybe we should take this into consideration.

Reply:
Good catch, thx! Will fix that.